src/cargo/util/graph.rs

```diff
@@ -1,10 +1,10 @@
 use std::fmt;
 use std::hash::Hash;
-use std::collections::hash_set::{HashSet, Iter};
-use std::collections::hash_map::{HashMap, Keys};
+use std::collections::hash_set::HashSet;
+use std::collections::hash_map::{HashMap, Keys, Iter};
 
-pub struct Graph<N> {
-    nodes: HashMap<N, HashSet<N>>
+pub struct Graph<N, L> {
+    nodes: HashMap<N, HashMap<N, HashSet<L>>>
```

Comment:
This may be best expressed as `HashMap<N, L>` and then we externally have something like `Vec<ActivationCond>`